src/test/java/picard/sam/AddOATagTest.java

```diff
@@ -0,0 +1,142 @@
+        clpOutput.deleteOnExit();
+
+        runAddOATag(inputSam, clpOutput, interval_list, null);
+
+        validateBamHasOA(clpOutput, interval_list, null);
+    }
+
+    @Test(dataProvider = "testOverwriteData")
+    public void testOverWrite(final File inputSam, final File interval_list, Boolean overWriteTag) throws IOException {
+        final File firstPassOutput = File.createTempFile("FirstPassAddOATag", ".bam");
+        firstPassOutput.deleteOnExit();
+        final File secondPassOutput = File.createTempFile("SecondPassAddOATag", ".bam");
+        secondPassOutput.deleteOnExit();
+
+        // make first pass bam that only has one value per OA tag
+        runAddOATag(inputSam, firstPassOutput, interval_list, true);
+
+        // make bam we want to test the overwrite option with
+        runAddOATag(firstPassOutput, secondPassOutput, interval_list, overWriteTag);
+
+        validateBamHasOA(secondPassOutput, interval_list, overWriteTag);
+    }
+
+    private void runAddOATag(final File inputSam, final File output, final File intervalList, Boolean overwriteTag) throws IOException {
+        final ArrayList<String> args = new ArrayList<String>(){
+            {
+                add("INPUT=" + inputSam);
+                add("OUTPUT=" + output);
+            }
+        };
+        if (intervalList != null) {
+            args.add("INTERVAL_LIST=" + intervalList);
+        }
+        if (overwriteTag != null) {
+            args.add("OVERWRITE_TAG=" + overwriteTag);
+        }
+        AddOATag addOATag = new AddOATag();
+        Assert.assertEquals(addOATag.instanceMain(args.toArray(new String[args.size()])), 0, "Running addOATag did not succeed");
+    }
+
+    private void validateOATAG(final File testSam, final File truthSam) {
+        final ArrayList<String> truthOAValues = new ArrayList<>();
+        final ArrayList<String> testOAValues = new ArrayList<>();
+
+        SAMRecordIterator iterator = SamReaderFactory.makeDefault().open(truthSam).iterator();
+        while (iterator.hasNext()){
+            SAMRecord rec = iterator.next();
+            truthOAValues.add(rec.getStringAttribute("OA"));
+        }
+
+        iterator = SamReaderFactory.makeDefault().open(testSam).iterator();
+        while (iterator.hasNext()){
+            SAMRecord rec = iterator.next();
+            testOAValues.add(rec.getStringAttribute("OA"));
+        }
+
+        Assert.assertTrue(truthOAValues.equals(testOAValues));
+    }
+
+    private void validateBamHasOA(final File input, final File intervalList, Boolean overwriteTag) {
+        final SAMRecordIterator iterator = SamReaderFactory.makeDefault().open(input).iterator();
+
+        OverlapDetector<Interval> detector = AddOATag.getOverlapDetectorFromIntervalListFile(intervalList, 0, 0);
+        while (iterator.hasNext()){
+            SAMRecord rec = iterator.next();
+            if (!rec.getReadUnmappedFlag() && (detector == null || detector.overlapsAny(rec))){
```

Comment:
I still prefer that all the validation is done with real output files, as this for example does not catch the problem that I pinpoint about unmapped pairs with paired mate (and thus, it is unclear if the behaviour is expected).